test/parallel_api/iterator/input_data_sweep_usm_allocator.pass.cpp

```diff
@@ -0,0 +1,127 @@
+        }
+
+        { //std::vector using usm host allocator
+            TestUtils::usm_data_transfer<sycl::usm::alloc::shared, T> copy_out(policy.queue(), n);
+            oneapi::dpl::counting_iterator<int> counting(0);
+            // usm_host allocator std::vector
+            sycl::usm_allocator<T, sycl::usm::alloc::host> q_alloc{policy.queue()};
+            std::vector<T, decltype(q_alloc)> host_data_vec(n, q_alloc);
+            //test all modes / wrappers
+
+            //Only test as source iterator for permutation iterator if we can expect it to work 
+            // (if the vector implementation distiguishes its iterator for this type)
+            wrap_recurse<__recurse, 0, /*__read =*/true, /*__reset_read=*/true, /*__write=*/true,
+                         /*__check_write=*/true, /*__usable_as_perm_map=*/true, /*__usable_as_perm_src=*/
+                         TestUtils::__vector_impl_distinguishes_usm_allocator_from_default<
+                             decltype(host_data_vec.begin())>::value,
+                         /*__is_reversible=*/true>(
+                policy, host_data_vec.begin(), host_data_vec.end(), counting, copy_out.get_data(),
+                host_data_vec.begin(), copy_out.get_data(), counting, trash,
+                std::string("usm_host_alloc_vector<") + type_text + std::string(">"));
+        }
+    }
+    else
+    {
+        TestUtils::unsupported_types_notifier(policy.queue().get_device());
+    }
+}
+
+#endif //TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+    bool run_tests = false;
+#if TEST_DPCPP_BACKEND_PRESENT
+    constexpr size_t n = 10;
+
+    auto q = TestUtils::get_test_queue();
+
+    auto policy = TestUtils::make_new_policy<class Kernel1>(q);
+
+    auto policy1 = TestUtils::create_new_policy_idx<decltype(policy), 0>(policy);
+    auto policy2 = TestUtils::create_new_policy_idx<decltype(policy), 1>(policy);
+    auto policy3 = TestUtils::create_new_policy_idx<decltype(policy), 2>(policy);
+    auto policy4 = TestUtils::create_new_policy_idx<decltype(policy), 3>(policy);
+    auto policy5 = TestUtils::create_new_policy_idx<decltype(policy), 4>(policy);
+
+    // baseline with no wrapping
+    if (oneapi::dpl::usm_allocated_vector_iterators_supported_v<float>)
+    {
+        run_tests = true;
+        test<float, 0>(policy1, -666.0f, n, "float");
+    }
+    if (oneapi::dpl::usm_allocated_vector_iterators_supported_v<double>)
+    {
+        run_tests = true;
+        test<double, 0>(policy2, -666.0, n, "double");
+    }
+    if (oneapi::dpl::usm_allocated_vector_iterators_supported_v<uint64_t>)
+    {
+        run_tests = true;
+        test<std::uint64_t, 0>(policy3, 999, n, "uint64_t");
+    }
+    if (oneapi::dpl::usm_allocated_vector_iterators_supported_v<int32_t>)
+    {
+        run_tests = true;
+        // big recursion step: 1 and 2 layers of wrapping
+        test<std::int32_t, 2>(policy4, -666, n, "int32_t");
+    }
+    }
```

Comment:
I think there is an extra `}` here.

Reply:
It looks like you are right, so the more problematic thing is then, how are the tests passing...?

I'll take a look.

Reply:
I was able to catch this when testing one of my branches with this PR. In the per-commit CI with the DPCPP backend, I don't think this test is ever ran which is why it is not caught.

Reply:
Ah, i see.  we only run the sycl_iterator tests in the per commit CI for dpcpp.  OK, makes sense. 
Thanks for catching this.

Reply:
fixed. Thanks.  I've also rebased after #1438, which enables this test to succeed on linux.  I've removed the checks of the public utility as that was separated out from the mentioned PR.